### PR TITLE
Raid PR - Jest conversion to Vitest

### PR DESCRIPTION
**Pull request by JacobMGEvans**
Converting Test framework from Jest to Vitest: https://vitest.dev/guide/comparisons.html
Steps and guidance for those unfamiliar with Jest or Vitest can follow the guide: https://vitest.dev/guide/migration.html
- Other improvements to testing environment will be included (i.e. GH Action Workflow in initial commit)

### CODE REVIEW

**Comment by JacobMGEvans** [comment]
@Borrus-sudo I need you to run the workflows

File: packages/compiler/test/index.test.ts [hunk 1]
```diff
@@ -1,8 +1,45 @@
...
+function useCompileInTemp({
+  fromFilename,
+  toFilename = "index.js",
+}: {
+  fromFilename: string;
+  toFilename?: string;
+}) {
+  let result = undefined;
+  beforeEach(() => {
+    spawnSync("yarn workspace @tinypages/compiler build", { shell: true });
```
Comment (JacobMGEvans) [inline packages/compiler/test/index.test.ts:15]: This part is workspace specific but can be modified if you make more tests for other modules.

File: packages/compiler/test/index.test.ts [hunk 2]
```diff
@@ -1,8 +1,45 @@
+import { describe, it, expect, beforeEach, afterEach } from "vitest";
+import { spawnSync } from "child_process";
 import * as fs from "fs";
-import { compile } from "../out/index.js";
+import * as path from "path";
 
+function useCompileInTemp({
```
Comment (JacobMGEvans) [inline packages/compiler/test/index.test.ts:6]: The benefit of this is that you now have a testing ecosystem living in an ethereal state. That allows for tests to run without concern of modifying file structure, outputs, etc... Meaning you can also run really bad scenarios, cache things to share between tests etc...

**Comment by Borrus-sudo** [comment]
Cool! Should I merge it? The workflow has passed.

**Comment by JacobMGEvans** [comment]
Sure! I can say the Raid was simple enough and any further improvements can be another Raid if anyone wants to tackle this more on the testing coverage and frameworks side of things.

**Pull request merged**